### PR TITLE
more compiler cleanup [nfc]

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Review by Keno — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
What does "nfc" stand for? Near-field communication?

**Comment by Keno** [comment]
No functional changes

File: src/jloptions.c [hunk 1]
```diff
@@ -31,12 +31,12 @@ jl_options_t jl_options = { 0,    // quiet
                             -1,   // banner
                             NULL, // julia_bindir
                             NULL, // julia_bin
-                            NULL, // project
                             NULL, // cmds
                             NULL, // image_file (will be filled in below)
                             NULL, // cpu_target ("native", "core2", etc...)
                             0,    // nprocs
                             NULL, // machine_file
+                            NULL, // project
```
Comment (StefanKarpinski) [inline src/jloptions.c]: Thanks for the fix!

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/27945#issuecomment-403058732) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/afe8465b91c6a995b18577a2eaabec737e47e851/2ba2025_vs_feaee9e/report.md). cc @ararslan

**Comment by vtjnash** [comment]
I don't understand how this could have affected allocations for `sumcolon_view` and `sumrange_view `, and time for `["problem", "seismic"]`

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks("sumcolon_view" || "sumrange_view" || "seismic", vs = ":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/27945#issuecomment-403255562) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/aa0b1dcbd345d743f3958fbfcf83b1de514f26c3/330d857_vs_2245a04/report.md). cc @ararslan

**Comment by vtjnash** [comment]
Not precisely NFC anymore, since this uncovered optimizations that we thought were happening but weren’t, so @nanosoldier `runbenchmarks(ALL, vs = ":master")` with the fixes for those

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/27945#issuecomment-403677917) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/4bd63b6ad761299bce29f706a1eff2c1e4cec3df/1ec85fb_vs_d8f70a5/report.md). cc @ararslan

**Pull request merged**